shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/STSTests.hs

```diff
@@ -145,19 +147,19 @@ stsTests :: TestTree
 stsTests =
   testGroup
     "STS Tests"
-    [ testCase "CHAIN example 1 - empty block" $ testCHAINExample exEmptyBlock,
-      testCase "CHAIN example 2A - register stake key" $ testCHAINExample (ex2A),
-      testCase "CHAIN example 2B - delegate stake and create reward update" $ testCHAINExample (ex2B p),
-      testCase "CHAIN example 2C - new epoch changes" $ testCHAINExample (ex2C p),
-      testCase "CHAIN example 2D - second reward update" $ testCHAINExample (ex2D p),
-      testCase "CHAIN example 2E - nonempty pool distr" $ testCHAINExample (ex2E p),
-      testCase "CHAIN example 2F - decentralized block" $ testCHAINExample (ex2F p),
-      testCase "CHAIN example 2G - prelude to the first nontrivial rewards" $ testCHAINExample (ex2G p),
-      testCase "CHAIN example 2H - create a nontrivial rewards" $ testCHAINExample (ex2H p),
-      testCase "CHAIN example 2I - apply a nontrivial rewards" $ testCHAINExample (ex2I p),
-      testCase "CHAIN example 2J - drain reward account and deregister" $ testCHAINExample (ex2J p),
-      testCase "CHAIN example 2K - stage stake pool retirement" $ testCHAINExample (ex2K p),
-      testCase "CHAIN example 2L - reap stake pool" $ testCHAINExample (ex2L p),
+    [ testCase "CHAIN example - empty block" $ testCHAINExample exEmptyBlock,
+      testCase "CHAIN example - pool lifetime - register stake key" $ testCHAINExample poolLifetime1,
+      testCase "CHAIN example - pool lifetime - delegate stake and create reward update" $ testCHAINExample poolLifetime2,
+      testCase "CHAIN example - pool lifetime - new epoch changes" $ testCHAINExample poolLifetime3,
+      testCase "CHAIN example - pool lifetime - second reward update" $ testCHAINExample poolLifetime4,
+      testCase "CHAIN example - pool lifetime - nonempty pool distr" $ testCHAINExample poolLifetime5,
+      testCase "CHAIN example - pool lifetime - decentralized block" $ testCHAINExample poolLifetime6,
+      testCase "CHAIN example - pool lifetime - prelude to the first nontrivial rewards" $ testCHAINExample poolLifetime7,
+      testCase "CHAIN example - pool lifetime - create a nontrivial rewards" $ testCHAINExample poolLifetime8,
+      testCase "CHAIN example - pool lifetime - apply a nontrivial rewards" $ testCHAINExample poolLifetime9,
+      testCase "CHAIN example - pool lifetime - drain reward account and deregister" $ testCHAINExample poolLifetime10,
+      testCase "CHAIN example - pool lifetime - stage stake pool retirement" $ testCHAINExample poolLifetime11,
+      testCase "CHAIN example - pool lifetime - reap stake pool" $ testCHAINExample poolLifetime12,
```

Comment:
Might be nicer to collect these as a group and export that group from `PoolLifetime`, since then if we add to them we only have to change in one file.

Reply:
great idea!

Reply:
I've made a `poolLifetime` test group, now located in `PoolLifetime`. And group the other examples as well, including the multisig ones (which are not currently enabled). Each example was previously being used twice, once to check that the expectation was correct, and once for preservation of ada. I've now combined these two tests into one, so that `testCHAINexample` now does both and the test group can do everything.